org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Finally.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+import static org.jacoco.core.test.validation.targets.Stubs.ex;
+import static org.jacoco.core.test.validation.targets.Stubs.f;
+import static org.jacoco.core.test.validation.targets.Stubs.nop;
+
+public class Finally {
+
+	/**
+	 * <pre>
+	 *   InputStream in = null;
+	 *   try {
+	 *     in = ...;
+	 *     ...
+	 *   } finally {
+	 *     if (in != null) {
+	 *       in.close();
+	 *     }
+	 *   }
+	 * </pre>
+	 */
+	private static void example(boolean t) {
+		Object in = null;
+		try {
+			in = open(t);
+		} finally { // $line-example.0$
+			if (in != null) { // $line-example.1$
+				nop(); // $line-example.2$
+			} // $line-example.3$
+		} // $line-example.4$
+	}
+
+	private static Object open(boolean t) {
+		ex(t);
+		return new Object();
+	}
+
+	private static void breakStatement() {
+		for (int i = 0; i < 1; i++) { // $line-breakStatement.for$
+			try {
+				if (f()) {
+					break; // $line-breakStatement$
+				}
+			} finally {
+				nop("finally"); // $line-breakStatement.1$
+			} // $line-breakStatement.2$
+		}
+	}
+
+	private static void catchNotExecuted() {
+		try {
+			nop("try");
+		} catch (Exception e) { // $line-catchNotExecuted$
+			nop("catch");
+		} finally { // $line-catchNotExecuted.0$
+			nop("finally"); // $line-catchNotExecuted.1$
+		} // $line-catchNotExecuted.2$
+	}
+
+	private static void emptyCatch() {
+		try {
+			nop("try");
+		} catch (Exception e) { // $line-emptyCatch$
+			// empty
+		} finally { // $line-emptyCatch.0$
+			nop("finally"); // $line-emptyCatch.1$
+		} // $line-emptyCatch.2$
+	}
+
+	private static void nested() {
+		try {
+			nop();
+		} finally {
+			try {
+				nop();
+			} finally { // $line-nested.0$
+				nop(); // $line-nested.1$
+			} // $line-nested.2$
+		}
+	}
+
+	private static void emptyTry() {
+		try {
+			// empty
+		} finally { // $line-emptyTry.0$
+			nop(); // $line-emptyTry.1$
+		} // $line-emptyTry.2$
+	}
+
+	private static void alwaysCompletesAbruptly() {
+		try {
+			nop();
+		} finally { // $line-alwaysCompletesAbruptly.0$
+			return; // $line-alwaysCompletesAbruptly.1$
```

Comment:
@Godin Here I get a warning in Eclipse, maybe annotate method with `@SuppressWarnings("finally")`?

Reply:
@marchof added this suppression, however warnings and suppressions are incredibly inconsistent in IDEs, e.g. even after addition I see warnings in this target file and in some other target files in IntelliJ. Ideally, I think, that our main code should not contain warnings and suppressions, but should be possible to write whatever in targets without bothering with warnings/suppressions.

Reply:
We could simply add `@SuppressWarnings("all")` to our validation targets. Should work for all IDEs.